test/api/capi/test_capi_arrow.cpp

```diff
@@ -114,4 +114,32 @@ TEST_CASE("Test arrow in C API", "[capi]") {
 		duckdb_destroy_arrow(&arrow_result);
 		duckdb_destroy_prepare(&stmt);
 	}
+
+	// test scan arrow
+	{
+		REQUIRE(duckdb_prepare(tester.connection, "SELECT CAST($1 AS BIGINT)", &stmt) == DuckDBSuccess);
+		REQUIRE(stmt != nullptr);
+		REQUIRE(duckdb_bind_int64(stmt, 1, 42) == DuckDBSuccess);
+		REQUIRE(duckdb_execute_prepared_arrow(stmt, nullptr) == DuckDBError);
+		REQUIRE(duckdb_execute_prepared_arrow(stmt, &arrow_result) == DuckDBSuccess);
+
+		ArrowSchema *arrow_schema = new ArrowSchema();
+		REQUIRE(duckdb_query_arrow_schema(arrow_result, (duckdb_arrow_schema *)&arrow_schema) == DuckDBSuccess);
+		REQUIRE(string(arrow_schema->format) == "+s");
+
+		ArrowArray *arrow_array = new ArrowArray();
+		REQUIRE(duckdb_query_arrow_array(arrow_result, (duckdb_arrow_array *)&arrow_array) == DuckDBSuccess);
+		REQUIRE(arrow_array->length == 1);
+
+		REQUIRE(duckdb_arrow_array_scan(tester.connection, "foo_table", (duckdb_arrow_schema)arrow_schema,
+		                                (duckdb_arrow_array)arrow_array) == DuckDBSuccess);
```

Comment:
As you said in the PR there is no need to extensive testing, but I still would like to see tests with larger than vector sizes (i.e., > 2048 elements), empty streams, all null values, and more unhappy path testing (e.g., double release, consuming the same stream twice, etc.)

Reply:
Also validate the results are correct

Reply:
Exactly, that's what I meant with empty streams, now that I understand your intention with the release calls, you can skip these. You can also skip the double stream consumption, since this is mostly to verify that the stream data is properly released, but if that's the responsibility of the caller, I don't think that it makes much sense to test it.

Thanks for all your work on this, highly appreciated! :-)

Reply:
@pdet I updated the test like so:
```
// test scan arrow
	{
		// Create a table with a `value` column from 1..1M.
		int num_buffers = 2, size = STANDARD_VECTOR_SIZE * num_buffers;
		const auto logical_types = duckdb::vector<LogicalType> {LogicalType(LogicalTypeId::INTEGER)};
		const auto column_names = duckdb::vector<string> {"value"};

		ArrowSchema *arrow_schema = new ArrowSchema();
		duckdb::ArrowConverter::ToArrowSchema(arrow_schema, logical_types, column_names, "");

		ArrowAppender appender(logical_types, size);
		Allocator allocator;

		auto data_chunks = std::vector<DataChunk>(num_buffers);
		for (int i = 0; i < num_buffers; i++) {
			auto data_chunk = &data_chunks[i];
			data_chunk->Initialize(allocator, logical_types, STANDARD_VECTOR_SIZE);
			data_chunk->SetCardinality(STANDARD_VECTOR_SIZE);
			for (int row = 0; row < STANDARD_VECTOR_SIZE; row++) {
				data_chunk->SetValue(0, row, duckdb::Value(i));
			}

			appender.Append(*data_chunk, 0, data_chunk->size(), data_chunk->size());
		}


		ArrowArray *arrow_array = new ArrowArray();
		*arrow_array = appender.Finalize();

		// Create temporary view.
		string view_name = "foo_table";
		REQUIRE(duckdb_arrow_array_scan(tester.connection, view_name.c_str(), (duckdb_arrow_schema)arrow_schema,
		                                (duckdb_arrow_array)arrow_array) == DuckDBSuccess);

		// Get created view from DB.
		auto get_query = "SELECT * FROM " + view_name;
		REQUIRE(duckdb_prepare(tester.connection, get_query.c_str(), &stmt) == DuckDBSuccess);
		REQUIRE(stmt != nullptr);
		REQUIRE(duckdb_execute_prepared_arrow(stmt, &arrow_result) == DuckDBSuccess);

		// Recover array from results.
		ArrowArray *out_array = new ArrowArray();
		REQUIRE(duckdb_query_arrow_array(arrow_result, (duckdb_arrow_array *)&out_array) == DuckDBSuccess);
		REQUIRE(out_array->length == size);
		out_array->release(out_array);
		delete out_array;

		arrow_schema->release(arrow_schema);
		delete arrow_schema;
		arrow_array->release(arrow_array);
		delete arrow_array;

		duckdb_destroy_arrow(&arrow_result);
		duckdb_destroy_prepare(&stmt);
	}
```

It needed a couple fixes in my code (not pushed these yet, but happy to if it helps/doesn't clutter the PR). The issue I now run into is that in `arrow_conversion.cpp`, for function `void ArrowTableFunction::ArrowToDuckDB`:

```
auto &array = *scan_state.chunk->arrow_array.children[arrow_array_idx];
```

`children` seems to have been destroyed/released 😢 any ideas as to why/what I'm doing incorrectly in my setup?

Reply:
@pdet Tests now include:
- Empty arrow
- Arrow twice as big as `STANDARD_VECTOR_SIZE` with read assertion
- Releases and destruction as expected from consumer

Please let me know if this looks good to you know!